helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/OptimalAssignment.java

```diff
@@ -50,22 +56,80 @@ public void updateAssignments(ClusterModel clusterModel) {
         .forEach(node -> _optimalAssignment.put(node, new ArrayList<>(node.getAssignedReplicas())));
   }
 
+  /**
+   * The coefficient of variation (CV) is a statistical measure of the dispersion of data points in
+   * a data series around the mean.
+   * The coefficient of variation represents the ratio of the standard deviation to the mean, and it
+   * is a useful statistic for comparing the degree of variation from one data series to another,
+   * even if the means are drastically different from one another.
+   * It's used as a tool to evaluate the "evenness" of an partitions to instances assignment
+   * @return a multi-dimension CV keyed by capacity key
+   */
+  public Map<String, Double> getCoefficientOfVariationAsEvenness() {
+    List<AssignableNode> instances = new ArrayList<>(_optimalAssignment.keySet());
+    Map<String, List<Integer>> usages = new HashMap<>();
+    for (AssignableNode instance : instances) {
+      Map<String, Integer> capacityUsage = instance.getCapacityUsage();
+      for (String key : capacityUsage.keySet()) {
+        usages.computeIfAbsent(key, k -> new ArrayList<>()).add(capacityUsage.get(key));
+      }
+    }
+
+    return usages.entrySet().stream()
+        .collect(Collectors.toMap(Map.Entry::getKey, e -> getCoefficientOfVariation(e.getValue())));
+  }
+
+  /**
+   * Compare this round of calculated assignment compared to the last assignment and get the
+   * partition movements count
+   * @param baseAssignment The base assignment (could be best possible/baseline assignment)
+   * @return a simple cumulative count of total movements where differences of movements in terms of
+   *         location, size, etc is ignored
+   */
+  public int getTotalPartitionMovements(Map<String, ResourceAssignment> baseAssignment) {
```

Comment:
Commented in the javadoc, the purpose of the method is to allow caller to pass any resource assignments (baseline or bestpossible) and it will calculate the movements